src/Source/Filter/HtmlFilter.php

```diff
@@ -258,6 +258,10 @@ function ($match) {
      */
     private function isIgnoredTag(?string $name): bool
     {
+        if ($name === null) {
+            return false;
```

Comment:
I think the tag should be ignored if its malformed. So this should return `true`.
Or else you will get a `array access on null` in https://github.com/mekras/php-speller/blob/629e86d8cc5b709924e92e3a4084d9629dc40351/src/Source/Filter/HtmlFilter.php#L123

Reply:
Thanks @icanhazstring . 
Actually if the tag is ignored, the result of the filter becomes `foo/             ` because the state is not restored and the text after the character `>` is replaced by spaces. Do  you have any idea to solve it?

Reply:
The base problem here I think is: The script "thinks" it is parsing a closing tag with `>` but it shouldn't. Because it never started one using `<`.

Maybe there is a total underlying problem here with the parsing method.
So I would go with your initial solution as it is the more correct one.

So your case with `$tagName !== null` should be in here.
Maybe I can think of some other solution some day :D

Reply:
Thanks a lot for your help

Reply:
Thanks for fixing :)